Geometry/HcalAlgo/plugins/DDHCalAngular.cc

```diff
@@ -32,37 +38,44 @@ void DDHCalAngular::initialize(const DDNumericArguments & nArgs,
   n           = int (nArgs["n"]);
   startCopyNo = int (nArgs["startCopyNo"]);
   incrCopyNo  = int (nArgs["incrCopyNo"]);
-  LogDebug("HCalGeom") << "DDHCalAngular debug: Parameters for positioning-- "
-		       << n << " copies in " << rangeAngle/CLHEP::deg 
-		       << " from " << startAngle/CLHEP::deg << "\tShifts " 
-		       << shiftX << ", " << shiftY 
-		       << " along x, y axes; \tZoffest " << zoffset
-		       << "\tStart and inremental copy nos " << startCopyNo 
-		       << ", " << incrCopyNo;
-
+#ifdef EDM_ML_DEBUG
+  edm::LogVerbatim("HCalGeom") << "DDHCalAngular: Parameters for positioning "
+			       << "-- " << n << " copies in " 
+			       << convertRadToDeg(rangeAngle) << " from " 
+			       << convertRadToDeg(startAngle) << "\tShifts " 
+			       << shiftX << ", " << shiftY 
+			       << " along x, y axes; \tZoffest " << zoffset
+			       << "\tStart and inremental copy nos " 
+			       << startCopyNo << ", " << incrCopyNo;
+#endif
   rotns       = sArgs["RotNameSpace"];
   idNameSpace = DDCurrentNamespace::ns();
   childName   = sArgs["ChildName"]; 
+#ifdef EDM_ML_DEBUG
   DDName parentName = parent().name(); 
-  LogDebug("HCalGeom") << "DDHCalAngular debug: Parent " << parentName 
-		       << "\tChild " << childName << "\tNameSpace "
-		       << idNameSpace << "\tRotation Namespace " << rotns;
+  edm::LogVerbatim("HCalGeom") << "DDHCalAngular: Parent " << parentName 
+			       << "\tChild " << childName << "\tNameSpace "
+			       << idNameSpace << "\tRotation Namespace "
+			       << rotns;
+#endif
 }
 
 void DDHCalAngular::execute(DDCompactView& cpv) {
 
   double dphi   = rangeAngle/n;
-  double phi    = startAngle;
+  double phix   = startAngle;
   int    copyNo = startCopyNo;
+  double theta  = 90._deg;
 
   for (int ii=0; ii<n; ii++) {
-
-    double phideg = phi/CLHEP::deg;
-    int    iphi;
-    if (phideg > 0) iphi = int(phideg+0.1);
-    else            iphi = int(phideg-0.1);
-    if (iphi >= 360) iphi   -= 360;
-    phideg = iphi;
+    double phideg = convertRadToDeg(phix);
+    int    iphi   = std::lround(phideg);
+    if (iphi >= 360) {
+      iphi   -= 360;
+      phideg  = iphi;
+      phix    = convertDegToRad(phideg);
```

Comment:
This line creates unnecessary differences. The original code does not round `phi`, as far as I can see. This line rounds it to the nearest degree.
I do not see any reason to round these angle values to the nearest degree. It was not done to `phi` previously. Why do it now?

Reply:
In the previous code it was using phideg*CLHEP::deg which was doing the rounding. The rounding is indeed needed.

Reply:
In fact if I keep phi but use something equivalent to phideg*CLHEP::deg I still see some differences in some of the angles. You cannot eliminate them. In fact here the translation and rotation matrix are consistent. In other case they are not

Reply:
CLHEP::deg = pi/180.0. It is a double.  `phideg*CLHEP::deg` converts phi in degrees to radians. There is no rounding. All arithmetic is double.
I am still not understanding the changes in this PR. The tiny differences you are reporting do not seem capable of causing thousands of differences. I have to fly to CERN Friday, so it may be a few days before I can puzzle over this PR some more.

Reply:
One of the motivations for removing CLHEP is to eliminate unnecessary conversions from radians to degrees. Please see how this sort of code is handled in [DDMuonAngular.cc.](https://github.com/cms-sw/cmssw/blob/6383740652b50371bd88dcd809fe8ab28e9c3510/Geometry/MuonCommonData/plugins/DDMuonAngular.cc#L70)
I don't think there is any need to convert angles to degrees except for debug messages.

Avoiding these conversions and roundings may eliminate the differences in the comparison tests.